content/docs/guides/auth/ALTS.md

```diff
@@ -0,0 +1,249 @@
+account of a GCE VM can be set or changed using
+[gCloud command](https://cloud.google.com/sdk/gcloud/reference/compute/instances/set-service-account)
+or via
+[GCP console](https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances#using).
+
+Google Cloud Platform issues an ALTS credential for each service account running
+on the GCE VM. The ALTS credentials are securely located in the hypervisor. The
+private key of an ALTS credential is not accessible to the VM and the
+application. The session keys used for end-to-end encryption are exposed to the
+gRPC stack. Google Cloud Platform fully manages the ALTS credentials, including
+certificate issuing, certificate rotation, and certification revocation.
+
+### gRPC Client with ALTS Transport Security Protocol
+
+gRPC clients can use ALTS credentials to connect to servers. The following
+provides examples of how to set up such clients in all supported gRPC languages.
+
+#### C++
+
+```cpp
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/security/credentials.h>
+
+using grpc::experimental::AltsCredentials;
+using grpc_impl::experimental::AltsCredentialsOptions;
+
+auto creds = AltsCredentials(AltsCredentialsOptions());
+std::shared_ptr<grpc::Channel> channel = CreateChannel(server_address, creds);
+```
+
+#### Java
+
+```java
+import io.grpc.alts.AltsChannelBuilder;
+import io.grpc.ManagedChannel;
+
+ManagedChannel managedChannel =
+    AltsChannelBuilder.forTarget(serverAddress).build();
+```
+
+#### Go
+
+```go
+import (
+  "google.golang.org/grpc"
+  "google.golang.org/grpc/credentials/alts"
+)
+
+altsTC := alts.NewClientCreds(alts.DefaultClientOptions())
+conn, err := grpc.Dial(serverAddr, grpc.WithTransportCredentials(altsTC))
+```
+
+#### Python
+
+```python
+import grpc
+
+channel_creds = grpc.alts_channel_credentials()
+channel = grpc.secure_channel(address, channel_creds)
+```
+
+### gRPC Server with ALTS Transport Security Protocol
+
+gRPC servers can use ALTS credentials to allow clients to connect to them. The
+following provides examples of how to set up such servers in all supported gRPC
+languages.
+
+#### C++
+
+```cpp
+#include <grpcpp/security/server_credentials.h>
+#include <grpcpp/server.h>
+#include <grpcpp/server_builder.h>
+
+using grpc::experimental::AltsServerCredentials;
+using grpc_impl::experimental::AltsServerCredentialsOptions;
+
+grpc::ServerBuilder builder;
+builder.RegisterService(&service);
+auto creds = AltsServerCredentials(AltsServerCredentialsOptions());
+builder.AddListeningPort("[::]:<port>", creds);
+std::unique_ptr<Server> server(builder.BuildAndStart());
+```
+
+#### Java
+
+```java
+import io.grpc.alts.AltsServerBuilder;
+import io.grpc.Server;
+
+Server server = AltsServerBuilder.forPort(<port>)
+    .addService(new MyServiceImpl()).build().start();
+
+```
+
+#### Go
+
+```go
+import (
+  "google.golang.org/grpc"
+  "google.golang.org/grpc/credentials/alts"
+)
+
+altsTC := alts.NewClientCreds(alts.DefaultClientOptions())
```

Comment:
This seems to be the client code :)

A server example can be found here:
https://github.com/grpc/grpc-go/blob/master/examples/features/encryption/ALTS/server/main.go

Reply:
Good catch. Copy&paste mistake :(